### PR TITLE
DOC: signal.firwin: emphasise cutoff freq is half-amplitude

### PR DESCRIPTION
**Pull request by zpotthoff**
#### Reference issue

Closes DOC: Specify cut-off frequency in firwin as half-ampltude vs half-power https://github.com/scipy/scipy/issues/20638

#### What does this implement/fix?

Edits documentation for `scipy.signal.firwin` to emphasize that the cutoff frequency is half-amplitude, different from `scipy.signal.iirdesign` which is half-power.

### CODE REVIEW

**Comment by gerazov** [comment]
Maybe also add information in `scipy.signal.iirdesign`?

**Comment by zpotthoff** [comment]
Hi @gerazov, I adjusted to add this information into `scipy.signal.iirdesign` and rephrased `scipy.signal.firwin`, is this more what you're looking for?

**Comment by gerazov** [comment]
Yes - looks good! :pray:

**Comment by larsoner** [comment]
Is there a good spot to add the code from #20638? Maybe a little `.. note::` in https://docs.scipy.org/doc/scipy/tutorial/signal.html#filter-design ?

**Comment by zpotthoff** [comment]
Hi @larsoner, how does this look?

File: doc/source/tutorial/signal.rst [hunk 1]
```diff
@@ -610,6 +610,35 @@ stop-band attenuation of :math:`\approx 60` dB.
...
+        >>> plt.figure()
+        >>> plt.plot(f, h_amp)
+        >>> plt.grid()
+
+        >>> b, a = sig.iirfilter(15, 2500, btype="low", fs=fs)
+        >>> f, h_fft = sig.freqz(b, a, fs=fs)
+        >>> h_amp = 20 * np.log10(np.abs(h_fft))
+        >>> plt.plot(f, h_amp)
+        >>> plt.axis([2100, 2900, -10, 2])
+        >>> plt.legend(["FIR", "IIR"])
```
Comment (larsoner) [inline doc/source/tutorial/signal.rst:640]: Let's modernize this a bit -- class-based is preferred over global state MPL nowadays, constrained layouts generally look better, and we might as well label the axes:
```suggestion
        >>> _, ax = plt.subplots(layout="constrained")
        >>> ax.plot(f, h_amp, label="FIR")
        >>> ax.grid(True)

        >>> b, a = sig.iirfilter(15, 2500, btype="low", fs=fs)
        >>> f, h_fft = sig.freqz(b, a, fs=fs)
        >>> h_amp = 20 * np.log10(np.abs(h_fft))
        >>> ax.plot(f, h_amp, label="IIR")
        >>> ax.set(xlim=[2100, 2900], ylim=[-10, 2])
        >>> ax.set(xlabel="Frequency (Hz)", ylabel="Amplitude (dB)")
```

Comment (larsoner) [reply doc/source/tutorial/signal.rst:640]: ... I know some of the other code uses the old interface, but we might as well make incremental improvement with the new code here!

Comment (zpotthoff) [reply doc/source/tutorial/signal.rst]: That makes sense, is there anything else in here that needs to be changed?

**Comment by zpotthoff** [comment]
Hi @larsoner, thanks for the feedback! Just checking back in, is there anything else that needs to be changed?

**Comment by larsoner** [comment]
CI failure is unrelated and looks great, thanks @zpotthoff !

**Pull request merged**